pyproject.toml

```diff
@@ -97,7 +98,7 @@ exclude = [
 ]
 
 [tool.ruff.lint.isort]
-known-first-party = ["resolvelib"]
+required-imports = ["from __future__ import annotations"]
```

Comment:
There's also the [`FA` rules](https://docs.astral.sh/ruff/rules/#flake8-future-annotations-fa), which might make this setting redundant :)

Reply:
In the past I found FA rules didn't always enable `from __future__ import annotations` when I expected, but looking at this PR it seems like all the additional `from __future__ import annotations`  aren't stictly necessary.

I've updated to use FA instead, it makes the PR a lot less noisey. Thanks.